R/thematic.R

```diff
@@ -178,18 +227,30 @@ thematic_off <- function() {
 #' thematic_off()
 #'
 #' thematic_with_theme(
-#'   scales::show_col(thematic_get_mixture(seq(0, 1, by = 0.1))),
-#'   "darkblue", "skyblue"
+#'   thematic_theme("darkblue", "skyblue"),
+#'   scales::show_col(thematic_get_mixture(seq(0, 1, by = 0.1)))
 #' )
 #'
-thematic_with_theme <- function(expr, ...) {
-  old <- thematic_on(...)
-  on.exit(thematic_set_theme(old), add = TRUE)
-  force(expr)
+thematic_with_theme <- function(theme, expr) {
+  old_theme <- thematic_set_theme(theme)
+  on.exit(thematic_set_theme(old_theme), add = TRUE)
+  expr <- rlang::enquo(expr)
+  result <- withVisible(rlang::eval_tidy(expr))
+  if (result$visible) print(result$value)
+  invisible(result$value)
```

Comment:
Add comment about why this is here and maybe hyperlink to similar code used in other places (e.g., `htmltools::capturePlot()`)